setup.py

```diff
@@ -57,13 +57,14 @@ def read(fname):
 
 aws_deps = ['boto3']
 gcp_deps = ['google-cloud-storage']
+asb_deps = ['azure-storage-blob', 'azure-common', 'azure-core']
 
-all_deps = install_requires + aws_deps + gcp_deps
+all_deps = install_requires + aws_deps + gcp_deps + asb_deps
 
 setup(
     name='smart_open',
     version=__version__,
-    description='Utils for streaming large files (S3, HDFS, GCS, gzip, bz2...)',
+    description='Utils for streaming large files (S3, HDFS, GCS, ASB, gzip, bz2...)',
```

Comment:
Is ASB common terminology here? I never heard of Azure blobs referred to as ASB.

Google shows me ASB stands for:

1. Advanced System Bus
2. Azure Service Bus

Neither seems relevant here.

Reply:
> Is ASB common terminology here? I never heard of Azure blobs referred to as ASB.
> 
> Google shows me ASB stands for:
> 
> 1. Advanced System Bus
> 2. Azure Service Bus
> 
> Neither seems relevant here.

Annoyingly for smart-open it looks like Microsoft has chosen to deviate from Google and Amazon by not creating a protocol for blobs. Instead they follow [the convention ](https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#resource-uri-syntax)`http://myaccount.blob.core.windows.net/mycontainer/myblob`.

Reply:
That's fine. Let's call it `azure` then, or `azureblob` / `AzureBlob`. Nobody knows what ASB is.